test/consistent-function-scoping.js

```diff
@@ -356,6 +390,47 @@ ruleTester.run('consistent-function-scoping', rule, {
 			`,
 			errors: [createError('function \'doBar\'')]
 		},
+		{
+			code: outdent`
+				const doFoo = function() {
+					function doBar(bar) {
+						return bar;
+					}
+				};
+			`,
+			errors: [createError('function \'doBar\'')]
+		},
+		{
+			code: outdent`
+				const doFoo = function() {
+					const doBar = function(bar) {
+						return bar;
+					};
+				};
+			`,
+			errors: [createError('function')]
```

Comment:
This function should has name `doBar`, https://github.com/mysticatea/eslint-utils/pull/14

Reply:
Is this PR blocked by that?

Reply:
No, we can improve message later.